### ISSUE TITLE
pytorch SimpleTrainingExample failed

### ISSUE THREAD

**Issue by lordk911** [intel-analytics/analytics-zoo#2812]
OS: centos 7.4
zoo: 0.8.1
python env : conda 4.8.4 , python 3.7,  torch 1.6.0+cpu,  torchvision 0.7.0+cpu, pyspark 2.4.3 
test code : https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/zoo/examples/pytorch/train/SimpleTrainingExample.py

error msg:
```
ERROR:root:Exception while sending command.
Traceback (most recent call last):
  File "/data/sk/env/zoo/lib/python3.7/site-packages/py4j/java_gateway.py", line 1159, in send_command
    raise Py4JNetworkError("Answer from Java side is empty")
py4j.protocol.Py4JNetworkError: Answer from Java side is empty

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/sk/env/zoo/lib/python3.7/site-packages/py4j/java_gateway.py", line 985, in send_command
    response = connection.send_command(command)
  File "/data/sk/env/zoo/lib/python3.7/site-packages/py4j/java_gateway.py", line 1164, in send_command
    "Error while receiving", e, proto.ERROR_ON_RECEIVE)
py4j.protocol.Py4JNetworkError: Error while receiving
Traceback (most recent call last):
  File "testorch.py", line 66, in <module>
    nnClassifierModel = classifier.fit(df)
  File "/data/sk/env/zoo/lib/python3.7/site-packages/pyspark/ml/base.py", line 132, in fit
    return self._fit(dataset)
  File "/data/sk/env/zoo/lib/python3.7/site-packages/pyspark/ml/wrapper.py", line 295, in _fit
    java_model = self._fit_java(dataset)
  File "/data/sk/env/zoo/lib/python3.7/site-packages/pyspark/ml/wrapper.py", line 292, in _fit_java
    return self._java_obj.fit(dataset._jdf)
  File "/data/sk/env/zoo/lib/python3.7/site-packages/py4j/java_gateway.py", line 1257, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/data/sk/env/zoo/lib/python3.7/site-packages/pyspark/sql/utils.py", line 63, in deco
    return f(*a, **kw)
  File "/data/sk/env/zoo/lib/python3.7/site-packages/py4j/protocol.py", line 336, in get_return_value
    format(target_id, ".", name))
py4j.protocol.Py4JError: An error occurred while calling o83.fit
ERROR:py4j.java_gateway:An error occurred while trying to connect to the Java server (127.0.0.1:44104)
Traceback (most recent call last):
  File "/data/sk/env/zoo/lib/python3.7/site-packages/py4j/java_gateway.py", line 929, in _get_connection
    connection = self.deque.pop()
IndexError: pop from an empty deque
```

**Comment by qiuxin2012** [comment]
This SimpleTrainingExample.py example only support torch 1.1.0
You can try this [mnist example](https://github.com/intel-analytics/analytics-zoo/tree/master/pyzoo/zoo/examples/pytorch/train/mnist), but you should update your az to [latest nightly build](https://analytics-zoo.github.io/master/#PythonUserGuide/install/#install-the-latest-nightly-build-wheels-for-pip).

**Comment by lordk911** [comment]
thanks, I've tried https://github.com/intel-analytics/analytics-zoo/tree/master/pyzoo/zoo/examples/pytorch/train/mnist .
When I run spark-on-yarn client mode I got error with : 
```
Caused by: jep.JepException: jep.JepException: <class 'zipimport.ZipImportError'>: can't decompress data; zlib not available
	at com.intel.analytics.zoo.common.PythonInterpreter$.threadExecute(PythonInterpreter.scala:87)
	at com.intel.analytics.zoo.common.PythonInterpreter$.exec(PythonInterpreter.scala:96)
```

and I also wonder to know why I have put the data to hdfs but the program also download it to local.

**Comment by lordk911** [comment]
```
(/data/sk/env/zoo) [devuser@xxx.host zoo]$ jep
>>> from java.lang import System
>>> System.out.println('hello, world')
hello, world
>>> import zipimport
>>> type(zipimport)
<class 'module'>
>>> import zlib
>>> type(zlib)
<class 'module'>
>>> 

```

**Comment by qiuxin2012** [comment]
> thanks, I've tried https://github.com/intel-analytics/analytics-zoo/tree/master/pyzoo/zoo/examples/pytorch/train/mnist .
> When I run spark-on-yarn client mode I got error with :
> 
> ```
> Caused by: jep.JepException: jep.JepException: <class 'zipimport.ZipImportError'>: can't decompress data; zlib not available
> 	at com.intel.analytics.zoo.common.PythonInterpreter$.threadExecute(PythonInterpreter.scala:87)
> 	at com.intel.analytics.zoo.common.PythonInterpreter$.exec(PythonInterpreter.scala:96)
> ```
> 
> and I also wonder to know why I have put the data to hdfs but the program also download it to local.
Could you try to install zlib first?
https://unix.stackexchange.com/questions/291737/zipimport-zipimporterror-cant-decompress-data-zlib-not-available/334103

**Comment by lordk911** [comment]
> > thanks, I've tried https://github.com/intel-analytics/analytics-zoo/tree/master/pyzoo/zoo/examples/pytorch/train/mnist .
> > When I run spark-on-yarn client mode I got error with :
> > ```
> > Caused by: jep.JepException: jep.JepException: <class 'zipimport.ZipImportError'>: can't decompress data; zlib not available
> > 	at com.intel.analytics.zoo.common.PythonInterpreter$.threadExecute(PythonInterpreter.scala:87)
> > 	at com.intel.analytics.zoo.common.PythonInterpreter$.exec(PythonInterpreter.scala:96)
> > ```
> > 
> > 
> > and I also wonder to know why I have put the data to hdfs but the program also download it to local.
> 
> Could you try to install zlib first?
> https://unix.stackexchange.com/questions/291737/zipimport-zipimporterror-cant-decompress-data-zlib-not-available/334103

I've install zlib by : sudo yum -y install zlib*  on all the nodemanagers

**Comment by qiuxin2012** [comment]
The same error with #2699.

> thanks, I've tried https://github.com/intel-analytics/analytics-zoo/tree/master/pyzoo/zoo/examples/pytorch/train/mnist .
> When I run spark-on-yarn client mode I got error with :
> 
> ```
> Caused by: jep.JepException: jep.JepException: <class 'zipimport.ZipImportError'>: can't decompress data; zlib not available
> 	at com.intel.analytics.zoo.common.PythonInterpreter$.threadExecute(PythonInterpreter.scala:87)
> 	at com.intel.analytics.zoo.common.PythonInterpreter$.exec(PythonInterpreter.scala:96)
> ```
> 
> and I also wonder to know why I have put the data to hdfs but the program also download it to local.

Because this pytorch dataset and dataloader doesn't support reading from hdfs.
Could add my wechat 18321938931, we are grad to help you find the solution, #2699 meets the same error.

**Comment by qiuxin2012** [comment]
Fixed in #2828 #2864
